tt_eager/tt_dnn/op_library/backward/backward_ops.cpp

```diff
@@ -21,22 +21,31 @@ namespace tt {
 namespace tt_metal {
 
 
-std::vector<Tensor> _addalpha_bw(const Tensor& grad, const Tensor& input, const Tensor& other, float alpha, const MemoryConfig& output_mem_config) {
-    std::vector<Tensor> grad_tensor;
-    grad_tensor.emplace_back(grad);
-    Tensor grad_b = mul_unary(grad, alpha, output_mem_config);
-    grad_tensor.emplace_back(grad_b);
-
-    return grad_tensor;
+std::vector<std::optional<Tensor>>  _addalpha_bw(const Tensor& grad, const Tensor& input, const Tensor& other, float alpha, const MemoryConfig& output_mem_config, const std::vector<bool> are_required_outputs, std::vector<std::optional<Tensor>> output_tensor) {
```

Comment:
Is it possible to return a std::optional<Tensor> here and not return a std::vector?

Reply:
there are 2 possible output tensors to be returned, input_grad and other_grad and say both are required, in that case we wouldn't be able to use <std::optional<Tensor> ?